### PR TITLE
feat:340 add invoke transaction

### PR DESCRIPTION
**Pull request by drspacemn**
RPC endpoint for adding `INVOKE` transactions

# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature [x]
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Can only invoke transactions via extrinsic 

Resolves: #340 

## What is the new behavior?

- `starknet_addInvokeTransaction`

## Does this introduce a breaking change?

No

### CODE REVIEW

File: crates/client/rpc-core/src/utils.rs [hunk 1]
```diff
@@ -61,3 +64,23 @@ pub(crate) fn _to_deprecated_entrypoint_by_type(
     });
     DeprecatedEntryPointsByType { constructor, external, l_1_handler }
 }
+
+pub fn to_invoke_tx(tx: BroadcastedInvokeTransaction) -> Result<InvokeTransaction> {
```
Comment (EvolveArt) [inline crates/client/rpc-core/src/utils.rs:68]: can we switch this to a `TryInto` trait impl ?

**Review by EvolveArt — COMMENTED** [review]
LGTM apart from the small change

**Comment by EvolveArt** [comment]
> LGTM apart from the small change

Oh and please update `docs/starknet_features_compatibility.md` too

File: crates/client/rpc-core/src/utils.rs (hunk 1 above)
Comment (drspacemn) [reply crates/client/rpc-core/src/utils.rs:46]: I can't add a `TryInto` outside the crate of the type delaraction (i.e. rpc-core)

I tried to add a `TryInto` in `primitives/transaction` but the issue is `starknet-providers` doesn't expose a no-std feature

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**